### PR TITLE
Static import test

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Remove "static import" under omerojava. Remove previous usage of JUnit, used prior to the switch to testng
One file (``RenderingEngineTest``) has not been modified, it will be cleaned up in https://github.com/openmicroscopy/openmicroscopy/pull/4753

# Testing this PR
make sure the integration tests are still green.

# Related reading

Ref: https://trello.com/c/9nZfJ6Pi/132-import-static-cleanup
and discussion on https://github.com/openmicroscopy/openmicroscopy/pull/4711

### CODE REVIEW

File: components/tools/OmeroJava/test/integration/AbstractServerTest.java [hunk 1]
```diff
@@ -921,10 +915,9 @@ protected RepositoryPrx getManagedRepository() throws ServerError {
     protected void assertExists(IObject obj) throws Exception {
         IObject copy = iQuery.find(obj.getClass().getSimpleName(), obj.getId()
                 .getValue());
-        assertNotNull(
+        Assert.assertNotNull(copy,
                 String.format("%s:%s", obj.getClass().getName(), obj.getId()
-                        .getValue())
-                        + " is missing!", copy);
+                        .getValue())+ " is missing!");
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/AbstractServerTest.java]: space before `+`, also below

File: components/tools/OmeroJava/test/integration/BfPixelBufferTest.java [hunk 2]
```diff
@@ -6,9 +6,6 @@
  */
 package integration;
 
-import static org.testng.AssertJUnit.assertEquals;
-import static org.testng.AssertJUnit.assertTrue;
-import static org.testng.AssertJUnit.fail;
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/BfPixelBufferTest.java]: remove a blank line

File: components/tools/OmeroJava/test/integration/ClientUsageTest.java [hunk 3]
```diff
@@ -6,12 +6,6 @@
  */
 package integration;
 
-import static omero.rtypes.rbool;
-import static omero.rtypes.rstring;
-import static org.testng.AssertJUnit.assertEquals;
-import static org.testng.AssertJUnit.assertFalse;
-import static org.testng.AssertJUnit.assertTrue;
-import static org.testng.AssertJUnit.fail;
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/ClientUsageTest.java]: remove a blank

File: components/tools/OmeroJava/test/integration/DeleteServiceTest.java [hunk 4]
```diff
@@ -1406,17 +1399,17 @@ public void testDeleteObjectWithSharableAnnotations() throws Exception {
...
-                assertTrue(annotationIds.size() > 0);
+                Assert.assertTrue(annotationIds.size() > 0);
                 sql = "select i from Annotation as i where i.id in (:ids)";
                 l = iQuery.findAllByQuery(sql, param);
                 if (values[j]) {
-                    assertEquals(l.size(), annotationIds.size(), l.size());
+                    Assert.assertEquals(l.size(), annotationIds.size(), l.size());
                 } else {
-                    assertEquals(l.toString(), 0, l.size());
+                    Assert.assertEquals(0, l.size(),l.toString());
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/DeleteServiceTest.java]: space after comma

File: components/tools/OmeroJava/test/integration/MetadataServiceTest.java [hunk 5]
```diff
@@ -2058,55 +2050,55 @@ public void testLoadImportLog() throws ServerError, CannotCreateSessionException
         /* test import log retrieval by fileset ID */
 
         logFiles = iMetadata.loadLogFiles(Fileset.class.getName(), ImmutableList.<Long>of());
-        assertTrue(logFiles.isEmpty());
+        Assert.assertTrue(logFiles.isEmpty());
 
         logFiles = iMetadata.loadLogFiles(Fileset.class.getName(), ImmutableList.of(fileset1Id));
-        assertTrue(logFiles.size() == 1);
+        Assert.assertTrue(logFiles.size() == 1);
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/MetadataServiceTest.java]: I was hoping for more `equals` :smiley:

File: components/tools/OmeroJava/test/integration/PixelsServiceTest.java [hunk 6]
```diff
@@ -238,43 +234,43 @@ public void testRetrievePixelsDescription() throws Exception {
...
-        assertTrue(pixels.getSizeT().getValue() == p.getSizeT().getValue());
-        assertTrue(pixels.getSizeZ().getValue() == p.getSizeZ().getValue());
-        assertTrue(pixels.getSizeC().getValue() == p.getSizeC().getValue());
-        assertTrue(pixels.sizeOfChannels() == ModelMockFactory.DEFAULT_CHANNELS_NUMBER);
-        assertTrue(p.sizeOfChannels() == pixels.sizeOfChannels());
-        assertTrue(pixels.getPhysicalSizeX().getValue() == p.getPhysicalSizeX()
+        Assert.assertNotNull(p);
+        Assert.assertTrue(pixels.getSizeX().getValue() == p.getSizeX().getValue());
+        Assert.assertTrue(pixels.getSizeY().getValue() == p.getSizeY().getValue());
+        Assert.assertTrue(pixels.getSizeT().getValue() == p.getSizeT().getValue());
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/PixelsServiceTest.java]: `assertEquals`

Comment (jburel) [inline components/tools/OmeroJava/test/integration/PixelsServiceTest.java]: I have modified some tests but obviously not all.

**Comment by jburel** [comment]
replace some more assertTrue
hopefully the tests will stay green

**Comment by mtbc** [comment]
It's just the double-lines I objected to. :smiley:

File: components/tools/OmeroJava/test/integration/AbstractServerTest.java [hunk 7]
```diff
@@ -961,10 +953,9 @@ protected void assertAllExist(String className, Collection<Long> ids) throws Ser
     protected void assertDoesNotExist(IObject obj) throws Exception {
         IObject copy = iQuery.find(obj.getClass().getSimpleName(), obj.getId()
                 .getValue());
-        assertNull(
+        Assert.assertNull(copy,
                 String.format("%s:%s", obj.getClass().getName(), obj.getId()
-                        .getValue())
-                        + " still exists!", copy);
+                        .getValue())+ " still exists!");
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/AbstractServerTest.java]: space before '+'

**Comment by jburel** [comment]
since I was there, I went ahead and fixed few more ;-)

File: components/tools/OmeroJava/test/integration/MetadataServiceTest.java [hunk 8]
```diff
@@ -844,7 +844,7 @@ public void testLoadAnnotationsUsedNotOwned() throws Exception {
         List<IObject> result = f
                 .getMetadataService()
                 .loadAnnotationsUsedNotOwned(TagAnnotation.class.getName(), id1);
-        Assert.assertTrue(result.size() > 0);
+        Assert.assertNotEquals(result.size(), 0);
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/MetadataServiceTest.java]: This kind of thing might be more obvious as,
```java
Assert.assertFalse(result.isEmpty());
```
...? Perhaps just a matter of taste.

Comment (jburel) [inline components/tools/OmeroJava/test/integration/MetadataServiceTest.java]: probably nicer

**Comment by mtbc** [comment]
This is looking pretty good.

**Comment by jburel** [comment]
let's see tomorrow if I have not broken anything.

**Comment by mtbc** [comment]
Still looking good, just https://github.com/openmicroscopy/openmicroscopy/pull/4758#discussion_r72427539 to address.

**Comment by jburel** [comment]
@mtbc: fixed the remaining comment.
Cleaned up a bit extra lines, we are probably good now.

**Comment by jburel** [comment]
@mtbc: are you happy with the cleaning?

**Comment by mtbc** [comment]
:+1: Good to go!

**Comment by sbesson** [comment]
Thanks. What are the implications of this change for other organization repositories. We have lots of `import static org.testng...` in Bio-Formats tests for instance. Is this enforcing a policy/recommendation not to use `import static` for tests? If so, should we document it and what is the plan for rolling it out?

**Comment by jburel** [comment]
We have not discussed a policy, this is mainly to finish the work started few summer ago when cleaning the (old) tests. The newly created tests have been using the ``testng`` instead of ``JUnit`` (i.e. static import )
We can certainly add policy to the contributing page. Happy to do that. Still a lot to clean before making such page public.

**Pull request merged**